configure.ac

```diff
@@ -268,7 +268,7 @@ if test "${PROTOC}" != ""; then
     AC_CACHE_CHECK([${PROTOC} version],
 		   nl_cv_protoc_version,
 		   [
-		       nl_cv_protoc_version=`${PROTOC} --version 2>&1 | ${SED} -n -e 's/^.\{1,\}\([[0-9]]\{1,\}.[[0-9]]\{1,\}.[[0-9]]\{1,\}\)$/\1/gp'`
+		       nl_cv_protoc_version=`${PROTOC} --version 2>&1 | ${SED} -n -E 's/^[^0-9]*(([0-9]+\.)*[0-9]+).*/\1/p'`
```

Comment:
configure.ac is processed with m4 and '[' and ']' is a reserved character that will be consumed by m4. Consequently, it needs to be escaped as in the prior code with '[[' and ']]'. Does this work in practice, as-is?

Also, '-e' says "a script follows" whereas '-E' says "the script, no matter where it is, uses extended regular expressions". So, I think for this to work it would require '-E -e'. Does it work as-is?

Reply:
I've tested a slight tweak of:

```
${SED} -n -E -e 's/^[^0-9]*(([0-9]+\.)*[0-9]+).*/\1/p'
```

which seems to work well with both BSD and GNU sed. I believe that the '[' and ']' will still need to be escaped.

Reply:
Okay, I've pushed an update just now that I think will address your concerns. I'm in the process of running a test to see if the changes as proposed work for our use-case. I should be able to report back in 15-20 minutes.